platformics/codegen/tests/test_aggregate_queries.py

```diff
@@ -141,4 +143,122 @@ async def test_count_distinct_query(
         }
     """
     results = await gql_client.query(query, user_id=111, member_projects=[888])
-    assert results["data"]["samplesAggregate"]["aggregate"]["count"] == 2
+    assert results["data"]["samplesAggregate"]["aggregate"][0]["count"] == 2
+
+@pytest.mark.asyncio
+async def test_groupby_query(
+    sync_db: SyncDB,
+    gql_client: GQLTestClient,
+) -> None:
+    """
+    Test that we can perform a groupby query
+    """
+    with sync_db.session() as session:
+        SessionStorage.set_session(session)
+        SampleFactory.create_batch(2, owner_user_id=111, collection_id=888, collection_location="San Francisco, CA")
+        SampleFactory.create_batch(3, owner_user_id=111, collection_id=888, collection_location="Mountain View, CA")
+
+    query = """
+        query MyQuery {
+            samplesAggregate {
+                aggregate {
+                    groupBy {
+                        collectionLocation
+                    }
+                    count
+                }
+            }
+        }
+    """
+    results = await gql_client.query(query, user_id=111, member_projects=[888])
+    aggregate = results["data"]["samplesAggregate"]["aggregate"]
+    for group in aggregate:
+        if group["groupBy"]["collectionLocation"] == "San Francisco, CA":
+            assert group["count"] == 2
+        elif group["groupBy"]["collectionLocation"] == "Mountain View, CA":
+            assert group["count"] == 3
+
+@pytest.mark.asyncio
+async def test_groupby_query_with_nested_fields(
+    sync_db: SyncDB,
+    gql_client: GQLTestClient,
+) -> None:
+    """
+    Test that we can perform a groupby query with nested fields
+    """
+    with sync_db.session() as session:
+        SessionStorage.set_session(session)
+        sample_1 = SampleFactory(owner_user_id=111, collection_id=888, collection_location="San Francisco, CA")
+        sample_2 = SampleFactory(owner_user_id=111, collection_id=888, collection_location="Mountain View, CA")
+        SequencingReadFactory.create_batch(2, sample=sample_1, owner_user_id=sample_1.owner_user_id, collection_id=sample_1.collection_id)
+        SequencingReadFactory.create_batch(3, sample=sample_2, owner_user_id=sample_2.owner_user_id, collection_id=sample_2.collection_id)
+
+    query = """
+        query MyQuery {
+            sequencingReadsAggregate {
+                aggregate {
+                    groupBy {
+                        sample {
+                            collectionLocation
```

Comment:
Can we add a test that includes multiple group by fields, at different levels?

Reply:
Done!